js/ui/tag_box.js

```diff
@@ -1277,6 +1276,23 @@ var TagBox = SelectBox.inherit({
         delete this._tagTemplate;
     },
 
+    _removeDuplicates: function(from, what) {
```

Comment:
Looks like you fix a symptom, not a problem

Reply:
The old way used the array.removeDuplicates method which worked incorrectly for a remote store.